LetsDoIt.CustomValueTypes/src/LetsDoIt.CustomValueTypes/Email.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Net.Mail;
+using System.Text.Json.Serialization;
+
+namespace LetsDoIt.CustomValueTypes
+{
+
+    [JsonConverter(typeof(EmailJsonConverter))]
+    public readonly struct Email
+    {
+        private readonly string _value;
+
+        public Email(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value), "Email can not be null!");
+            }
+
+            if (!IsValid(value))
+            {
+                throw new ArgumentException("Invalid email address.", value);
+            }
+
+            _value = value.ToLowerInvariant();
+        }
+
+        public static bool IsValid(string email)
+        {
+            if (string.IsNullOrEmpty(email))
+            {
+                return false;
+            }
+
+            try
+            {
+                var mailAddress = new MailAddress(email);
+
+                return string.Equals(mailAddress.Address, email, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
why do we need this comparison?

Reply:
MailAdress.Adress has a validation of an email. We leverage this classes power. And normal MailAdress class accepts both mail address and author name but MailAdress.Adress only checks the email format. This method was suggested alternative to Regex validation, Here is a link:
 https://stackoverflow.com/questions/1365407/c-sharp-code-to-validate-email-address